### ISSUE TITLE
Load Vanilla Betaflight into Production Board

### ISSUE THREAD

**Issue by Dishoungh** [UARK-Quadcopter-Flight-Controller/betaflight#1]
Work on porting Betaflight vanilla into the production (green) board

**Comment by Dishoungh** [comment]
@sdw043 @srp001 Luckily, Betaflight does support the BMP280 and MPU6050. You can see them in /src/main/drivers/accgyro/accgyro_mpu6050.c and /src/main/drivers/barometer/barometer_bmp280.c

This is good. It seems we might not do too much with the betaflight vanilla code to port to our board since many of the common components have drivers already in them. To give you guys something to do, we can work on how we can use the Betaflight Configurator to communicate with the STM32 MCU and the peripherals. 

Simultaneously, I would like for us to look into how communication with the magnetometer is going to work. This isn't going to be a huge deal because the magnetometer isn't critical to the board's functionality. The magnetometer will be chained to the gyroscope to give it more accurate readings. 

![Mag](https://user-images.githubusercontent.com/47036723/111387482-c76f6d80-867b-11eb-8557-c54f25c472cb.png)
![Gyro](https://user-images.githubusercontent.com/47036723/111387483-c76f6d80-867b-11eb-8936-8ff8bffa980c.png)

As you can see in these pictures, the magnetometer isn't directly connected to the MCU and Betaflight might not like that. Unless there is a way that I haven't seen how to do that. I will swap boards with Zach possibly tomorrow. This weekend I will go back home and stay there for a little bit but I will have one of the boards. I will help you guys get Betaflight to one of these boards. 

I've been looking into these resources to get started:
[Betaflight Pin Remapping](https://www.propwashed.com/betaflight-resource-remapping/)
[Betaflight Configurator Tutorial](https://www.youtube.com/watch?v=oto_cuXRrE8)
[Betaflight - USB Flashing Document](https://github.com/UARK-Quadcopter-Flight-Controller/betaflight/blob/master/docs/USB%20Flashing.md)

**Comment by srp001** [comment]
Am I supposed to reply here? When would you like to do this? I can make time on any afternoon (I get off at 3) that is not Friday to work with you guys! I am busy all day/evening this Friday. Just let me know.

**Comment by Dishoungh** [comment]
@srp001 Sure, that's fine. On Friday, I'm going home so if you want to do this, tomorrow or Thursday is the best time. Tomorrow, Zach is planning on swapping boards again before I leave. I think meeting sometime after 4pm would be best for me.

**Comment by Dishoungh** [comment]
@srp001 Before we start, I highly recommend you cloning, creating a branch, and try pushing to the repo (do anything like slightly changing the README). Before I had issues so I'm not sure if you would see those same issues or not.

**Comment by zackheil** [comment]
There's a slight disconnect from the mission here from what I said yesterday. For this vanilla port, forget the BMP and the Gyro for now. The goal is to eliminate variables to keep track of where things break. Just configure the motor outputs to match the MCU page on the schematic and configure some GPIO pins (not 1 or 8) to be RC PWM receiver pins so we can get something basic to interact with the motors.

**Comment by Dishoungh** [comment]
@zackheil Okay, we'll see what we can do.

**Comment by zackheil** [comment]
How is this coming along? We are in a position where we can begin betaflight testing on our production board and get a first flight in and this would be nice to have in a few days

**Comment by Dishoungh** [comment]
@zackheil I'm currently working on testing the Magnetometer and helping Stephanie and Spencer with their tests afterwards. I'll try to get as much as I can today
